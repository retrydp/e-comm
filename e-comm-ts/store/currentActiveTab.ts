import { createSlice } from '@reduxjs/toolkit';

export interface CurrentActiveTab {
  activeTab: string;
}
const initialState: CurrentActiveTab = {
  activeTab: '',
};

const currentActiveTab = createSlice({
  name: 'currentTab',
  initialState,
  reducers: {
    change: (state, action) => {
      state.activeTab = action.payload;
    },
  },
});

export const { change } = currentActiveTab.actions;
export default currentActiveTab.reducer;
