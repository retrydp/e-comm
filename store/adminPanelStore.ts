import { createSlice } from '@reduxjs/toolkit';
import { ProductSchema, UserSchema } from '../utils/types';

type DataSchemas = ProductSchema[] | UserSchema[] | [];

interface AuthInitialState<T extends DataSchemas> {
  data: T;
  loading: boolean;
  error: string;
  loadingDelete: boolean;
  errorDelete: boolean;
  deleteErrorText: string;
}

const initialState: AuthInitialState<DataSchemas> = {
  data: [],
  loading: false,
  error: '',
  loadingDelete: false,
  errorDelete: false,
  deleteErrorText: '',
};

const adminPanelStore = createSlice({
  name: 'authStore',
  initialState,
  reducers: {
    fetchRequest: (state) => {
      state.loading = true;
      state.error = '';
    },
    fetchSuccess: (state, action) => {
      state.loading = false;
      state.data = action.payload;
      state.error = '';
    },
    fetchError: (state, action) => {
      state.loading = false;
    },
    deleteRequest: (state) => {
      state.loadingDelete = true;
    },
    deleteSuccess: (state, action) => {
      state.loadingDelete = false;
      state.deleteErrorText = '';
      state.data = action.payload;
    },
    deleteError: (state, action) => {
      state.loadingDelete = false;
      state.deleteErrorText = action.payload;
    },
  },
});

export const {
  fetchRequest,
  fetchSuccess,
  fetchError,
  deleteRequest,
  deleteSuccess,
  deleteError,
} = adminPanelStore.actions;

export default adminPanelStore.reducer;
