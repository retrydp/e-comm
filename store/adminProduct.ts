import { createSlice } from '@reduxjs/toolkit';

interface ApInitialState {
  adminProductLoading: boolean;
  adminProductLoadingAdd: boolean;
  adminProductErrorText: string;
  adminProductErrorAdd: string;
}

const initialState: ApInitialState = {
  adminProductLoading: false,
  adminProductLoadingAdd: false,
  adminProductErrorText: '',
  adminProductErrorAdd: '',
};

const adminProduct = createSlice({
  name: 'adminProduct',
  initialState,
  reducers: {
    adminProductUploadRequest: (state) => {
      state.adminProductLoading = true;
      state.adminProductErrorText = '';
    },
    adminProductUploadSuccess: (state) => {
      state.adminProductLoading = false;
      state.adminProductErrorText = '';
    },

    adminProductUploadError: (state, action) => {
      state.adminProductLoading = false;
      state.adminProductErrorText = action.payload;
    },
    adminProductAddRequest: (state) => {
      state.adminProductLoadingAdd = true;
      state.adminProductErrorAdd = '';
    },
    adminProductAddSuccess: (state) => {
      state.adminProductLoadingAdd = false;
      state.adminProductErrorAdd = '';
    },
    adminProductAddError: (state, action) => {
      state.adminProductLoadingAdd = false;
      state.adminProductErrorAdd = action.payload;
    },
  },
});

export const {
  adminProductUploadRequest,
  adminProductUploadSuccess,
  adminProductUploadError,
  adminProductAddError,
  adminProductAddSuccess,
  adminProductAddRequest,
} = adminProduct.actions;

export default adminProduct.reducer;
