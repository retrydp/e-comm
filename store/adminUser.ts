import { createSlice } from '@reduxjs/toolkit';

interface ApInitialState {
  adminUserLoading: boolean;
  adminUserErrorText: string;
}

const initialState: ApInitialState = {
  adminUserLoading: false,
  adminUserErrorText: '',
};

const adminUser = createSlice({
  name: 'adminUser',
  initialState,
  reducers: {
    adminUserEditRequest: (state) => {
      state.adminUserLoading = true;
      state.adminUserErrorText = '';
    },
    adminUserEditSuccess: (state) => {
      state.adminUserLoading = false;
      state.adminUserErrorText = '';
    },

    adminUserEditError: (state, action) => {
      state.adminUserLoading = false;
      state.adminUserErrorText = action.payload;
    },
  },
});

export const {
  adminUserEditRequest,
  adminUserEditSuccess,
  adminUserEditError,
} = adminUser.actions;

export default adminUser.reducer;
