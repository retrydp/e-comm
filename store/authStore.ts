import { createSlice } from '@reduxjs/toolkit';
import Cookies from 'js-cookie';
export interface AuthInitialState {
  userInfo: {
    token: string;
    _id: string;
    name: string;
    email: string;
    isAdmin: boolean;
  } | null;
}

const initialState: AuthInitialState = {
  userInfo: Cookies.get('userInfo')
    ? JSON.parse(Cookies.get('userInfo') || '')
    : null,
};

const authStore = createSlice({
  name: 'authStore',
  initialState,
  reducers: {
    userLogin: (state, action) => {
      state.userInfo = action.payload;
    },
    userLogout: (state) => {
      state.userInfo = null;
    },
  },
});

export const { userLogin, userLogout } = authStore.actions;

export default authStore.reducer;
