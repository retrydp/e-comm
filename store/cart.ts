import { createSlice } from '@reduxjs/toolkit';import { ProductSchema } from '../utils/types';
import Cookies from 'js-cookie';

interface CartProduct extends ProductSchema {
  count: number;
}

interface CartInitialState {
  products: CartProduct[];
}

const initialState: CartInitialState = {
  products: Cookies.get('cart')
    ? JSON.parse(Cookies.get('cart') as string)
    : [],
};

const cart = createSlice({
  name: 'cart',
  initialState,
  reducers: {
    addProduct: (state, action) => {
      const product = action.payload;
      const productIndex = state.products.findIndex(
        ({ slug }) => slug === product.slug
      );
      if (productIndex === -1) {
        state.products.push({ ...product, count: 1 });
      } else {
        state.products[productIndex].count += 1;
      }
    },
    incrementCount: (state, action) => {
      const product = action.payload;
      const productIndex = state.products.findIndex(
        ({ slug }) => slug === product.slug
      );
      state.products[productIndex].count += 1;
    },
    decrementCount: (state, action) => {
      const product = action.payload;
      const productIndex = state.products.findIndex(
        ({ slug }) => slug === product.slug
      );
      state.products[productIndex].count -=
        state.products[productIndex].count > 1 ? 1 : 0;
    },
    deleteProduct: (state, action) => {
      const product = action.payload;
      state.products = state.products.filter(
        ({ slug }) => slug !== product.slug
      );
    },
  },
});

export const { addProduct, incrementCount, decrementCount, deleteProduct } =
  cart.actions;

export default cart.reducer;
