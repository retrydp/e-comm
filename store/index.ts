import { configureStore } from '@reduxjs/toolkit';
import displayInterface from './displayInterface';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

const store = configureStore({
  reducer: displayInterface,
});

export default store;

export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();

export type RootState = ReturnType<typeof store.getState>;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
