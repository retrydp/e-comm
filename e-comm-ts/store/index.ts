import { configureStore } from '@reduxjs/toolkit';
import currentActiveTab from '../store/currentActiveTab';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
const store = configureStore({
  reducer: currentActiveTab,
});

export default store;

export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
export type RootState = ReturnType<typeof store.getState>;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
