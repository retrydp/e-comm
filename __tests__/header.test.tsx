import { render } from '@testing-library/react';
import Component from '../components/Header';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import React from 'react';
import store from '../store';

// jest.mock('next/router', () => ({
//   useRouter() {
//     return {
//       route: '/',
//       pathname: '',
//       query: { page: 1, quantity: 12 },
//       asPath: '',
//     };
//   },
// }));

describe('header test', () => {
  const tree = () =>
    render(
      <Provider store={store}>
        <Component />
      </Provider>
    );

  it('login button is present ', () => {
    const { getByText } = tree();
    expect(getByText('Log In')).toBeInTheDocument();
  });
  it('cart is present', () => {
    const { getByTestId } = tree();
    expect(getByTestId('ShoppingCartOutlinedIcon')).toBeInTheDocument();
  });
  it('favorites is present', () => {
    const { getByTestId } = tree();
    expect(getByTestId('FavoriteBorderIcon')).toBeInTheDocument();
  });
  it('menu is present', () => {
    const { getByTestId } = tree();
    expect(getByTestId('MenuIcon')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = tree();
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});
