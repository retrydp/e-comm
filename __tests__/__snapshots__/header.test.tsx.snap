// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header test matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1bn4qxc-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <a
        aria-label="Menu"
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1wm0gbs-MuiButtonBase-root-MuiButton-root"
        data-mui-internal-clone-element="true"
        href="/"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <div
        class="MuiBox-root css-i9gxme"
      />
      <div
        class="MuiBox-root css-ks2x85"
      >
        <a
          aria-label="Favorites"
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1wm0gbs-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          href="/favorites"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FavoriteBorderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          aria-label="User Cart"
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1wm0gbs-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          href="/cart"
          tabindex="0"
        >
          <span
            class="MuiBadge-root css-1c32n2y-MuiBadge-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingCartOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <span
              class="MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-invisible css-1pi4uwz-MuiBadge-badge"
            />
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          aria-label="Log in"
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          href="/login"
          tabindex="0"
        >
          Log In
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
